### DISCUSSION TITLE
How to use LegacyCoercingInputInterceptor in v22

### DISCUSSION THREAD

**Discussion by piotrblasiak** [graphql-java/graphql-java#3594] [Q&A] [answered]
It seems v21 uses this by default, but I can't figure out how to enable the LegacyCoercingInputInterceptor in v22.

**Comment by bbakerman** [comment, marked as answer]
The code that uses this callback is still in place in V22 - but it being on by default has changed

the value conversion code does

```
InputInterceptor inputInterceptor = graphqlContext.get(InputInterceptor.class);
```

So you need to place a `LegacyCoercingInputInterceptor` into the requests `graphql.GraphQLContext` under the key `graphql.execution.values.InputInterceptor` class.

It will then run that code during value conversion

The reason it was removed in the V22 breaking change version is that we wanted to move to a more spec  compliant value conversion and the breaking change version was the time to do it.   We knew it may hurt some people relying on old value shapes so we pre built this conversion callback to allow people to go back to the old ways if need be

**Reply by piotrblasiak** [reply]
Ok, thanks - setting it to LegacyCoercingInputInterceptor.migratesValues() seems to be working.

**Comment by pennal** [comment]
Since I just struggled for a while with this issue, I'll add my solution. I am using DGS 9.0.3, which in turn uses graphql-java. In order to do this at the global level in spring, you need to do the following:

```
@Component
public class CustomGraphQLContextContributor implements GraphQLContextContributor {

  @Override
  public void contribute(@NotNull GraphQLContext.Builder builder, @Nullable Map<String, ?> extensions, @Nullable DgsRequestData dgsRequestData) {
    builder.put(InputInterceptor.class, LegacyCoercingInputInterceptor.migratesValues());
  }

}
```

**Reply by dondonz** [reply]
Thanks for sharing!

**Comment by Boucaa** [comment]
If you're using this library in Spring Boot via spring-graphql, you can use it like this (in Kotlin):

```kotlin
class GraphQlInterceptor : WebGraphQlInterceptor {

    override fun intercept(request: WebGraphQlRequest, chain: WebGraphQlInterceptor.Chain): Mono<WebGraphQlResponse> {
        request.configureExecutionInput { _, builder ->
            builder.graphQLContext(mapOf(InputInterceptor::class.java to LegacyCoercingInputInterceptor.migratesValues()))
            builder.build()
        }
        return chain.next(request)
    }
}

@Configuration
class GraphQlInterceptorConfig {

    @Bean
    fun customGraphQlInterceptor(): WebGraphQlInterceptor {
        return GraphQlInterceptor()
    }
}

```
